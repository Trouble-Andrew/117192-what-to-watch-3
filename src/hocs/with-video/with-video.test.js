import React, {createRef} from "react";
import renderer from "react-test-renderer";
import withVideo from "./with-video.jsx";
import VideoPlayer from "../../components/video-player/video-player.jsx";

const MockComponentWrapped = withVideo(VideoPlayer);

it(`withVideo is rendered correctly`, () => {
  const videoRef = createRef();
  const tree = renderer.create((
    <MockComponentWrapped
      isPlaying={false}
      ref={videoRef}
      src={``}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
