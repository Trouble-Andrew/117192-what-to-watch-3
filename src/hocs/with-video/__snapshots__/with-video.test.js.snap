// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideo is rendered correctly 1`] = `
<video
  className="player__video"
  muted={true}
  poster="img/player-poster.jpg"
/>
`;
